src/System.CommandLine/Binding/BoundValue.cs

```diff
@@ -5,6 +5,7 @@ namespace System.CommandLine.Binding
 {
     public class BoundValue
     {
+        // ?? Why have an internal constructor on a public readonly class?
```

Comment:
We don't want people to create instances of this.

Reply:
It does not appear that BoundValue escape the context of System.CommandLine. I think we should deliberately consider the needs of any application models that need something other than to connect a ValueSource to a Option/Argument ahead of any evaluation. IOW, we may need this public in the future, but it does not appear that we need it public now. 

This is probably part of a broader discussion of checking our public API surface.